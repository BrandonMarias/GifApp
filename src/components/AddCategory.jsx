import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ handleFunction }) => {
    const [inputValue, setInputValue] = useState("one punch");

    const handleInputChange = ({ target }) => setInputValue(target.value);

    const handleOnSubmit = (event) => {
        event.preventDefault();
        if (!inputValue.trim()) return;
        handleFunction(inputValue.trim());
        setInputValue("");
    };

    return (
        <form onSubmit={handleOnSubmit} aria-label="form">
            <input
                type="text"
                placeholder="Bucar Gifs"
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    );
};

AddCategory.propTypes = {
    handleFunction: PropTypes.func.isRequired,
};
