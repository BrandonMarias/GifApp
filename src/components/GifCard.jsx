import propTypes from "prop-types";

export const GifCard = ({ title, img }) => {
    return (
        <div className="card">
            <img src={img} alt={title} />
            <p>{title}</p>
        </div>
    );
};

GifCard.propTypes = {
    title: propTypes.string.isRequired, // title of the gif
    img: propTypes.string.isRequired, // url of the gif image to be displayed
};
