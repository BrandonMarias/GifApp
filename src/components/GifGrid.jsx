
import { useFetchGifs } from "../hooks/useFetchGifs";
import {GifCard} from "./";



export const GifGrid = ({ category }) => {
    
    const [gifs, isLoading] = useFetchGifs(category)

    return (
        <>
            <h3>{category}</h3>
            {isLoading && <h2>Is loading...</h2>}
            <div className="card-grid">
                {gifs.map(({ id, title, img }) => (
                    <GifCard key={id} title={title} img={img} />
                ))}
            </div>
        </>
    );
};

