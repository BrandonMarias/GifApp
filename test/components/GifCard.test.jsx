import { render, screen } from "@testing-library/react";
import { GifCard } from "../../src/components/GifCard";

describe('test on <GifCard/>', () => { 
    const title = 'title';
    const img = 'https://media1.giphy.com/media/l0Hl6qZQQ7qQ/200.gif';
    test("should make a snapshot", () => {
        const {container} = render(<GifCard title="title" img="img" />);
        expect(container).toMatchSnapshot();
    });
    test("should show the title and image", () => {
    render(<GifCard title={title} img={img} />);
    // expect(document.querySelector('img').src).toBe(img);
    // expect(document.querySelector('img').alt).toBe(title);
    const { src, alt} = screen.getByRole('img');
    expect(src).toBe(img);
    expect(alt).toBe(title);

    } );
    test("should show the title in the paragraph", () => {
        render(<GifCard title={title} img={img} />);
        expect(screen.getByText(title)).toBeTruthy();    
    } );
 }) 