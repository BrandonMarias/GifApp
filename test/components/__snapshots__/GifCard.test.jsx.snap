// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifCard/> should make a snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="img"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
